src/Payment/PaymentApplication/Handlers/ProcessPaymentHandler.cs

```diff
@@ -21,17 +25,43 @@ public async Task<bool> Handle(ProcessPaymentCommand request, CancellationToken
 			if (payment is null || payment.Status != (int)PaymentStatus.InProgress)
 				return false;
 
+			string queueName;
+			Func<long, Task<bool>> task;
+			if (request.Complete)
+			{
+				queueName = KnownQueueNames.PaymentCompleted;
+				task = _paymentRepository.CompletePayment;
+			}
+			else
+			{
+				queueName = KnownQueueNames.PaymentFailed;
+				task = _paymentRepository.FailPayment;
+			}
+
 			bool result;
 			try
 			{
-				result = request.Complete ? await _paymentRepository.CompletePayment(request.Id)
-																	: await _paymentRepository.FailPayment(request.Id);
+				result = await task(request.Id);
 			}
-			catch (DbUpdateException)
+			catch
 			{
 				result = false;
 			}
 
+			if (result)
+			{
+				try
+				{
+					var details = await _paymentRepository.GetPaymentWithRelatedInfo(request.Id);
+					var notification = PaymentProcessedNotificationProducer.Get(details, request.Complete);
+					await _notificationsPublisher.PersistentPublish(notification, queueName);
+				}
+				catch 
+				{
+					// logging
+				}
```

Comment:
Ideally all operations like that should be in background, because you already have the **result** you need from this endpoint. For example you could implement some INotificationService with method SendNotfications(params...)
```
    public Task<Unit> SendNotfications(NotificationRequest request, CancellationToken cancellationToken)
    {
        _ = Task.Factory
            .StartNew(() => this.SendRequest(request), TaskCreationOptions.LongRunning)
            .Unwrap();

        return Task.FromResult(Unit.Value);
    }
```
SendRequest method will have try-catch inside, and Service will have publisher and all the repositories needed.
and in your code:
```
await this.notificationService.SendNotification(new NotificationRequest { type = PaymentComplete, queueName = ...})
```